bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java

```diff
@@ -0,0 +1,184 @@
+        super.encodeMessage(message);
+
+        final int pulsesByteLen = rawMessage.length - 5;
+        if (pulsesByteLen % 4 != 0) {
+            throw new RFXComException("Incorrect byte length for pulses - must be divisible by 4");
+        }
+
+        subType = SubType.fromByte(super.subType);
+        repeat = rawMessage[4];
+        pulses = new short[pulsesByteLen / 2];
+        ByteBuffer.wrap(rawMessage, 5, rawMessage.length - 5).asShortBuffer().get(pulses);
+    }
+
+    @Override
+    public byte[] decodeMessage() throws RFXComException {
+        if (pulses.length > 124) {
+            throw new RFXComMessageTooLongException("Longest payload according to RFXtrx SDK is 124 shorts.");
+        }
+
+        final int pulsesByteLen = pulses.length * 2;
+        byte[] data = new byte[5 + pulsesByteLen];
+
+        data[0] = (byte) (data.length - 1);
+        data[1] = RAW.toByte();
+        data[2] = subType.toByte();
+        data[3] = seqNbr;
+        data[4] = repeat;
+
+        ByteBuffer.wrap(data, 5, pulsesByteLen).asShortBuffer().put(pulses);
+
+        return data;
+    }
+
+    @Override
+    public String getDeviceId() {
+        return "RAW";
+    }
+
+    @Override
+    public State convertToState(String channelId, DeviceState deviceState) throws RFXComUnsupportedChannelException {
+        switch (channelId) {
+            case CHANNEL_RAW_MESSAGE:
+                return new StringType(HexUtils.bytesToHex(rawMessage));
+
+            case CHANNEL_RAW_PAYLOAD:
+                byte[] payload = new byte[pulses.length * 2];
+                ByteBuffer.wrap(payload).asShortBuffer().put(pulses);
+                return new StringType(HexUtils.bytesToHex(payload));
+
+            default:
+                throw new RFXComUnsupportedChannelException("Nothing relevant for " + channelId);
+        }
+    }
+
+    @Override
+    public void setSubType(SubType subType) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void setDeviceId(String deviceId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void convertFromState(String channelId, Type type) throws RFXComUnsupportedChannelException {
+        switch (channelId) {
+            case CHANNEL_RAW_MESSAGE:
+                if (type instanceof StringType) {
+                    // TODO: Check the raw message for validity (length, no more than 124 shorts, multiple of 4 bytes in
+                    // payload)
+                    throw new RFXComUnsupportedChannelException("Channel " + channelId + " inot yet implemented");
+                } else {
+                    throw new RFXComUnsupportedChannelException("Channel " + channelId + " does not accept " + type);
+                }
+
+            case CHANNEL_RAW_PAYLOAD:
+                if (type instanceof StringType) {
+                    // TODO: Check the payload for validity (no more than 124 shorts, multiple of 4 bytes
```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?

Reply:
This is really a TODO for when I implement tx, which is next. There is value in merging rx.